src/main/java/com/linecorp/armeria/server/http/file/MimeTypeUtil.java

```diff
@@ -88,7 +91,16 @@ static MediaType guessFromPath(String path) {
 
         final String extension = path.substring(dotIdx + 1).toLowerCase(Locale.US);
         final MediaType mediaType = EXTENSION_TO_MEDIA_TYPE.get(extension);
-        return mediaType != null ? mediaType : MediaType.parse(URLConnection.guessContentTypeFromName(path));
+        if (mediaType != null) {
+            return mediaType;
+        }
+        final String guessedContentType;
+        try {
+             guessedContentType = Files.probeContentType(Paths.get(path));
```

Comment:
It passes at the moment, but it involves service loaders which means a user can insert any probing implementations.